File: Pufferfish/SensorsData100HZStruct.cs
using System.Runtime.InteropServices;

namespace PufferFish;

[StructLayout(LayoutKind.Sequential, Pack = 1)]
internal class SensorsData100HZStruct
{
	public DataHeader header;

	public TimeStamp t;

	public SensorsData100HZ data;
}

File: ski-gear-python/data_classes/sensors_data_100hzStruct.py
from dataclasses import dataclass
from .data_header import DataHeader
from .timestamp import TimeStamp
from .sensors_data_100hz import SensorsData100HZ

@dataclass
class SensorsData100HZStruct:
    header: DataHeader
    t: TimeStamp
    data: SensorsData100HZ

    @classmethod
    def parse(cls, reader):

        header = DataHeader.parse(reader)
        timestamp = TimeStamp.parse(reader)
        data = SensorsData100HZ.parse(reader)

        return cls(header, timestamp, data)
